sympy/polys/rings.py

```diff
@@ -2531,6 +2531,124 @@ def compose(f, x, a=None):
 
         return poly
 
+
+    def coeff_wrt(self, sym, dg):
+        """
+        Coefficient of ``f`` with respect to ``x**deg``.
+
+        Treating ``f`` as a univariate polynomial in ``x`` this finds the
+        coefficient of ``x**deg`` as a polynomial in the other generators.
+
+        Parameters
+        ==========
+
+        p : sympy.Poly
+            The polynomial to compute the sparse coefficient for.
+        sym : int
+            The symbol to compute the coefficient for.
+        dg : int
+            The degree of the monomial to compute the coefficient for.
+
+        Returns
+        =======
+
+        sympy.Number
+            The sparse coefficient of the polynomial at the given symbol and degree.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> _, x, y, z = ring("x, y, z", ZZ)
+        >>> p = 2*x**4 + 3*y**4 + 10*z**2 + 3
+        >>> dg = 2
+        >>> sym = 2
+        >>> p.coeff_wrt(sym, dg)
+        10
+
+        """
+        p = self
+        terms = [(m, c) for m, c in p.iterterms() if m[sym] == dg]
+        monoms, coeffs = zip(*terms)
+        monoms = [m[:sym] + (0,) + m[sym + 1:] for m in monoms]
+        return p.ring.from_dict(dict(zip(monoms, coeffs)))
+
+
+    def prem2(self, g, x):
+        """
+        Computes the pseudo-remainder of the polynomial `f` with respect to `g`.
```

Comment:
Why is this called `prem2`? The existing `prem` method should be changed.

Reply:
The `x` parameter could be made optional like `x=None` and then by default it would return the `prem` as computed by the existing method (wrt the first generator).

Reply:
Then I also remove this one?
https://github.com/sympy/sympy/blob/eaa65dae74f43a84ae708fe0b91db215ea37d2f6/sympy/polys/rings.py#L2540-L2542

Reply:
Yes. It should be checked that the output of `f.prem(g)` is still the same as it was previously though. If there are no tests for the `prem` method then tests should be added.